test/integration/index_management.test.js

```diff
@@ -258,39 +271,106 @@ describe('Indexes', function () {
     }
   });
 
-  it('shouldCorrectlyDropIndexes', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
+  context('when dropIndexes succeeds', function () {
+    let collection;
 
-    test: function (done) {
-      var configuration = this.configuration;
-      var client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
-      var db = client.db(configuration.db);
-      db.createCollection('test_drop_indexes', function (err, collection) {
-        collection.insert({ a: 1 }, configuration.writeConcernMax(), function (err) {
-          expect(err).to.not.exist;
-          // Create an index on the collection
-          db.createIndex(
-            collection.collectionName,
-            'a',
-            configuration.writeConcernMax(),
-            function (err, indexName) {
-              test.equal('a_1', indexName);
-              // Drop all the indexes
-              collection.dropIndexes(function (err, result) {
-                test.equal(true, result);
+    beforeEach(async function () {
+      collection = await db.createCollection('test_drop_indexes');
+      await collection.insert({ a: 1 });
+      // Create an index on the collection
+      await db.createIndex(collection.collectionName, 'a');
+    });
 
-                collection.indexInformation(function (err, result) {
-                  test.ok(result['a_1'] == null);
-                  client.close(done);
-                });
-              });
-            }
-          );
+    afterEach(async function () {
+      await db.dropCollection('test_drop_indexes');
+    });
+
+    it('should return true and should no longer exist in the collection', async function () {
+      // Drop all the indexes
+      const result = await collection.dropIndexes();
+      expect(result).to.equal(true);
+
+      const res = await collection.indexInformation();
+      expect(res['a_1']).to.equal(undefined);
+    });
+  });
+
+  context('when dropIndexes fails', function () {
+    let collection;
+
+    beforeEach(async function () {
+      collection = await db.createCollection('test_drop_indexes');
+      await collection.insert({ a: 1 });
+      // Create an index on the collection
+      await db.createIndex(collection.collectionName, 'a');
+      /**@type {import('../tools/utils').FailPoint} */
+      await client
+        .db()
+        .admin()
+        .command({
+          configureFailPoint: 'failCommand',
+          mode: {
+            times: 1
+          },
+          data: {
+            failCommands: ['dropIndexes'],
+            errorCode: 91
+          }
         });
-      });
-    }
+    });
+
+    afterEach(async function () {
+      await db.dropCollection('test_drop_indexes');
+    });
+
+    it('should return false', {
+      metadata: {
+        requires: {
+          mongodb: '>4.0'
```

Comment:
```suggestion
          mongodb: '>=4.0'
```

I think this will pass on 4.0 servers too